src/repr/src/fixed_length.rs

```diff
@@ -0,0 +1,761 @@
+                    typ.scalar_type
+                ),
+            }
+        }
+        if let Some(d) = datum_iter.next() {
+            panic!(
+                "Row too wide: Found extra datum {:?} when expecting type {:?}",
+                *d.borrow(),
+                typ
+            );
+        }
+        result
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use chrono::{DateTime, NaiveDate, NaiveDateTime, NaiveTime, Utc};
+    use ordered_float::OrderedFloat;
+    use uuid::Uuid;
+
+    use crate::{
+        adt::{
+            date::Date,
+            interval::Interval,
+            mz_acl_item::{AclItem, MzAclItem},
+            timestamp::CheckedTimestamp,
+        },
+        fixed_length::{fixed_length, BytesN, FromRowByTypes, IntoRowByTypes},
+        scalar::AsColumnType,
+        ColumnType, Datum, Row, ScalarType, Timestamp,
+    };
+
+    /// This test takes inspiration from its namesake in `Row`, but instead
+    /// goes from `Row` to fixed-length and back.
+    #[mz_ore::test]
+    fn miri_test_round_trip() {
+        fn fixed_length_type(datum: Datum) -> Option<ColumnType> {
+            match datum {
+                Datum::False => Some(bool::as_column_type()),
+                Datum::True => Some(bool::as_column_type()),
+                Datum::Int16(_) => Some(i16::as_column_type()),
+                Datum::Int32(_) => Some(i32::as_column_type()),
+                Datum::Int64(_) => Some(i64::as_column_type()),
+                Datum::UInt16(_) => Some(u16::as_column_type()),
+                Datum::UInt32(_) => Some(u32::as_column_type()),
+                Datum::UInt64(_) => Some(u64::as_column_type()),
+                Datum::Float32(_) => Some(f32::as_column_type()),
+                Datum::Float64(_) => Some(f64::as_column_type()),
+                Datum::Date(_) => Some(Date::as_column_type()),
+                Datum::Time(_) => Some(NaiveTime::as_column_type()),
+                Datum::Timestamp(_) => Some(CheckedTimestamp::<NaiveDateTime>::as_column_type()),
+                Datum::TimestampTz(_) => Some(CheckedTimestamp::<DateTime<Utc>>::as_column_type()),
+                Datum::Interval(_) => Some(Interval::as_column_type()),
+                Datum::Uuid(_) => Some(Uuid::as_column_type()),
+                Datum::MzTimestamp(_) => Some(Timestamp::as_column_type()),
+                Datum::MzAclItem(_) => Some(MzAclItem::as_column_type()),
+                Datum::AclItem(_) => Some(AclItem::as_column_type()),
+                _ => None,
+            }
+        }
+
+        fn round_trip<const N: usize>(datums: Vec<Datum>, types: &[ColumnType]) {
+            let row = Row::pack(datums.clone());
+            let bytes_n = BytesN::<N>::from_row(row.clone(), types);
+
+            // The line below is intended to catch undefined bytes written to the
+            // fixed-length representation when run under miri.
+            println!("{:?}", bytes_n);
```

Comment:
Good point, it was indeed something copied from `Row` earlier, but it ended up that we do not use `unsafe` in the fixed-length implementation. Well, at least not directly; there is still unsafe code inside the implementation of `copy_from_slice`. Not sure if we should keep because of the latter or if we should just remove the `println!`?

Reply:
Oh, I wouldn't worry about `copy_from_slice`. You have to believe that you can trust the stdlib functions, otherwise how should you maintain your sanity?

Fine to remove it IMO, as long as we remember to put it back in if we ever add `unsafe` code to this module.

Reply:
Awesome, removed!